src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -185,23 +240,29 @@ private IServiceCollection BuildCommonServices(out AggregateException hostingSta
             var appEnvironment = PlatformServices.Default.Application;
             var contentRootPath = ResolveContentRootPath(_options.ContentRootPath, appEnvironment.ApplicationBasePath);
             var applicationName = _options.ApplicationName ?? appEnvironment.ApplicationName;
+            var hostingContext = new HostingContext();
 
             // Initialize the hosting environment
             _hostingEnvironment.Initialize(applicationName, contentRootPath, _options);
+            hostingContext.HostingEnvironment = _hostingEnvironment;
 
             var services = new ServiceCollection();
             services.AddSingleton(_hostingEnvironment);
 
+            var builder = new ConfigurationBuilder()
+                .SetBasePath(_hostingEnvironment.ContentRootPath)
+                .AddInMemoryCollection(_config.AsEnumerable());
+
+            _configureConfigurationBuilderDelegate?.Invoke(hostingContext, builder);
```

Comment:
Would it be confusing that that configuration is the one that's about to be overwritten? Also, is it okay to just call build on the builder to get an `IConfiguration` that contains the original config since we just copied it over? Would be unintuitive but not sure how common this use case is.